File: task_2/go/2(4).go
package main

import (
	"fmt"
	"math/rand"
	"time"
)

func getRandomChar() byte {
	const chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789!@#$%^&*()"
	return chars[rand.Intn(len(chars))]
}

func main() {
	rand.Seed(time.Now().UnixNano())

	const n = 10 // можно выбрать любое значение
	symbols := make([]byte, n)

	// Заполняем массив случайными символами
	for i := range symbols {
		symbols[i] = getRandomChar()
	}

	// Вывод начального массива
	fmt.Print("Исходный массив: [")
	for i := 0; i < n; i++ {
		fmt.Printf("%c", symbols[i])
		if i != n-1 {
			fmt.Print(", ")
		}
	}
	fmt.Println("]")

	// Сдвиг влево на 1, первый элемент — в конец
	first := symbols[0]
	copy(symbols[:n-1], symbols[1:])
	symbols[n-1] = first

	// Вывод модифицированного массива
	fmt.Print("Модифицированный массив: [")
	for i := 0; i < n; i++ {
		fmt.Printf("%c", symbols[i])
		if i != n-1 {
			fmt.Print(", ")
		}
	}
	fmt.Println("]")
}

File: task_2/cpp/2(4).cpp
#include <iostream>
#include <vector>
#include <ctime>
#include <cstdlib>

using namespace std;

char getRandomChar() {
    const string chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789!@#$%^&*()";
    return chars[rand() % chars.size()];
}

int main() {
    srand(static_cast<unsigned>(time(0)));

    const int n = 10; // можно выбрать любое значение
    vector<char> symbols(n);

    // Заполняем массив случайными символами
    for (char &c : symbols) {
        c = getRandomChar();
    }

    // Вывод начального массива
    cout << "Исходный массив: [";
    for (int i = 0; i < n; ++i) {
        cout << symbols[i];
        if (i != n - 1) cout << ", ";
    }
    cout << "]" << endl;

    // Сдвиг влево на 1, первый элемент — в конец
    char first = symbols[0];
    for (int i = 1; i < n; ++i) {
        symbols[i - 1] = symbols[i];
    }
    symbols[n - 1] = first;

    // Вывод модифицированного массива
    cout << "Модифицированный массив: [";
    for (int i = 0; i < n; ++i) {
        cout << symbols[i];
        if (i != n - 1) cout << ", ";
    }
    cout << "]" << endl;

    return 0;
}
